src/formatDate.js

```diff
@@ -50,7 +50,85 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+  let year, smallYear, month, day;
+  const resultArray = [];
+  let resultDate = '';
+
+  const fromArray = date.split(fromSeparator);
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY': {
+        year = fromArray[i];
+
+        break;
+      }
+
+      case 'YY': {
+        smallYear = fromArray[i];
+
+        break;
+      }
+
+      case 'MM': {
+        month = fromArray[i];
+
+        break;
+      }
+
+      case 'DD': {
+        day = fromArray[i];
+
+        break;
+      }
+    }
+  }
+
+  if (year !== undefined && year !== null) {
+    smallYear = year.slice(2);
+  }
+
+  if (smallYear !== undefined && smallYear !== null) {
+    if (smallYear < 30) {
+      year = 2000 + (+smallYear);
```

Comment:
```suggestion
      year = 2000 + Number(smallYear);
```
This suggestion is a bit more readable and convenient to maintain